src/app.ts

```diff
@@ -188,6 +190,48 @@ export type ContextData = {
 
 const fedifyKv = await KnexKvStore.create(client, 'key_value');
 
+let storage = new Storage();
+const bucketName = process.env.GCP_BUCKET_NAME;
+if (!bucketName) {
+    logging.error('GCP bucket name is not configured');
+    process.exit(1);
+}
+
+if (['staging', 'production'].includes(process.env.NODE_ENV || '')) {
+    try {
+        const [exists] = await storage.bucket(bucketName).exists();
+        if (!exists) {
+            throw new Error(`Bucket [${bucketName}] does not exist`);
+        }
+        logging.info('GCP bucket exists');
+    } catch (err) {
+        logging.error('Failed to verify GCP bucket {error}', {
+            error: err,
+        });
+        process.exit(1);
+    }
+}
+
+if (process.env.GCP_STORAGE_EMULATOR_HOST) {
+    storage = new Storage({
+        apiEndpoint: process.env.GCP_STORAGE_EMULATOR_HOST,
+        projectId: 'dev-project',
+        useAuthWithCustomEndpoint: false,
+        credentials: {
+            client_email: 'fake@example.com',
+            private_key: 'not-a-real-key',
+        },
+    });
+
+    const bucket = storage.bucket(bucketName);
+
+    const [exists] = await bucket.exists();
+    if (!exists) {
+        logging.info('Creating GCP bucket in fake-gcs');
+        await bucket.create(); //Create a bucket in fake-gcs
```

Comment:
fake-gcs-server starts with no buckets, so we create it in dev/test if it doesn't exist

Reply:
This shouldn't be in the application code if possible - it should be part of the docker stack if we can

We could introduce a script which runs as the container command, and it will execute when the container starts and ensure anything we need

Or we could commit to the repo a pre-populated data directory, and mount that as the volume 🤔